serpentTools/objects/xsdata.py

```diff
@@ -0,0 +1,120 @@
+"""Holds cross section data pertaining to Serpent xsplot output."""
+import numpy as np
+from matplotlib import pyplot
+
+from serpentTools import messages
+from serpentTools.objects import NamedObject, convertVariableName
+
+
+class XSData(NamedObject):
+    docParams = """name: str
+        Name of this material
+    metadata: dict
+        Dictionary with file metadata"""
+    __doc__ = """
+    Base class for storing cross section data an xsplot file
+
+    Parameters
+    ----------
+    {params:s}
+
+    Attributes
+    ----------
+
+    """.format(params=docParams)
```

Comment:
Why not just directly add `docParams` to the docstring? 

This approach is useful for deeper levels of abstraction like with the `DetectorBase`, `Detector`, and `SampledDetectors` from #93. If you intend on adding an `XsPlotSampler` to this PR* then this would absolutely make sense. Here, not a ton of benefit

*The approval of this PR is not dependent on adding such a sampler

Reply:
Yeah, I don't know. That's what I would have rather done, but I didn't know if you guys preferred to do docstrings like that for some reason, since that syle is present in a few other files.

On the XSPlotSampler, could you explain the significance of such a thing? I see that the "SampledDetector" can compute RMS error on the detector readings, and spread of error on detector readings. What exactly is this used for? I don't see why we wouldn't just include the plotting of uncertainty in the given energy grid, rather than make a whole new class to evaluate uncertainties.

Reply:
The reformatting of the docstrings using these class attribtues helps reduce the amount of retyping code when writing docstrings for subclasses. For example, the `Detector` and `SampledDetector` are *basically* the same object, just the latter utilizes data from many `Detector` instances as has some minor method tweaks. The class docstrings, listing the attributes of the two, are very similar, so I wrote in the `docAttrs` on the `DetectorBase` to easily list the shared attributes.

This isn't something that is beneficial for a one off container. If we eventually write an `XsPlotSampler`, then this type of docstring formatting would be useful. I don't know how beneficial an `XsPlotSampler` would be, to be honest. I'm not sure how widely used this feature is within the SERPENT community.